sql-to-dbsp-compiler/lib/sqllib/src/casts.rs

```diff
@@ -14,6 +14,29 @@ use rust_decimal::Decimal;
 const FLOAT_DISPLAY_PRECISION: usize = 6;
 const DOUBLE_DISPLAY_PRECISION: usize = 15;
 
+// Creates three cast functions based on an existing one
+macro_rules! cast_function {
+    ($result_name: ident, $result_type: ty, $type_name: ident, $arg_type: ty) => {
```

Comment:
I feel like this is a little confusing. 

I think it'd be better if it were: `result_name`, `type_name`, `arg_type`, `result_type`. 
It'd be consistent to the function definition and other macros that way.

Reply:
The order matches the order the types appear in the function name "cast_to_destination_source", so destination comes first in the list of arguments. I think I will leave this like this. But it's a good point.